### PR TITLE
Fixes lockers leaking out when changing matrices

### PR DESCRIPTION
**Pull request by Bod9001**
~~small fix needed It spawns the contents in when changing matrices~~
Shouldn't spawn initial contents when changing matrices  now

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/ObjectContainer.cs [hunk 1]
```diff
@@ -150,11 +150,11 @@ public void GatherObjects()
 			}
 		}
 
-		public IEnumerable<GameObject> GetStoredObjects()
+		public IEnumerable<GameObject> GetStoredObjects(bool OnlyInstantiated = false)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/ObjectContainer.cs]: All arguments variable names should be camelCase

**Comment by corp-0** [comment]
The intention of the bool argument in all of those 3 method isn't immediately clear

**Comment by corp-0** [comment]
Why did this start happening? I'm pretty sure it wasn't the case since ever

**Comment by Bod9001** [comment]
well, the way it's triggered is pretty self-explanatory,  when changing mattresses, it loops through all the contents and sets the matrix, however due to the default bool ,  if it hasn't been spawned in yet the contentsit instantiates it, and doesn't make it hidden,  I improved it more by make it so you can specify to only include stuff that is spawned inside the locker

File: UnityProject/Assets/Scripts/Objects/ObjectContainer.cs (hunk 1 above)
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/ObjectContainer.cs]:
```suggestion
		public IEnumerable<GameObject> GetStoredObjects(bool onlyInstantiated = false)
```

File: UnityProject/Assets/Scripts/Objects/ObjectContainer.cs [hunk 2]
```diff
@@ -150,11 +150,11 @@ public void GatherObjects()
...
 		{
-			if (initialContentsSpawned == false)
+			if (initialContentsSpawned == false && OnlyInstantiated == false)
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/ObjectContainer.cs]:
```suggestion
			if (initialContentsSpawned == false && onlyInstantiated == false)
```

**Review by NoooneyDude — APPROVED** [review]

**Review by ThatDan123 — APPROVED** [review]

**Comment by ThatDan123** [comment]
> Why did this start happening? I'm pretty sure it wasn't the case since ever

Guess not many people pushed lockers which hadn't already been opened before onto a different matrix. Rare bug.

**Pull request merged**